src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -21,11 +23,12 @@ public static void main(String[] args) {
         } catch (UnsupportedLookAndFeelException e) {
             e.printStackTrace();
         }
-        SwingUtilities.invokeLater(Main::new);
+        StartupParams params = args.length == 0 ? null : new StartupParams(args);
+        SwingUtilities.invokeLater(() -> new Main(params));
     }
 
     public static class NoOpBot {
-        public static void main(String[] args) {
+        public static void main(String[] args) throws IOException {
             ConfigManager.FORCE_NO_OP = true;
```

Comment:
This was a lazy way of passing what should've been a parameter to the API-side. Feel free to remove NoOpBot inner class and just make it one of the startup params `-no-op`, and treat it on the getAPI method in config manager